#10 - design list of meetups page

paulinaszkudlarek opened the pull request:
![image](https://user-images.githubusercontent.com/83571355/210322038-1714f84f-ef6f-4bfe-ab1d-f06dd6ea0f90.png)
Filtry:
![image](https://user-images.githubusercontent.com/83571355/210322161-17d73fa9-02e0-45ed-8113-91fa97478cbf.png)
Mobile:
![image](https://user-images.githubusercontent.com/83571355/210322326-23bd6f53-279b-46f3-8d86-bb4bec800bb8.png)
filtry na mobile:
![image](https://user-images.githubusercontent.com/83571355/210322398-97b4398b-07c2-456c-bceb-64069bc8198e.png)
![image](https://user-images.githubusercontent.com/83571355/210322561-6f23f96b-8e63-4403-88e2-a4030d37beff.png)

EwelinaLasowy reviewed (commented):
1. This button "Filtry" should be rounded as button "Sortuj domyślnie"
![image](https://user-images.githubusercontent.com/56546832/210238106-854d2418-e22e-4097-b918-ad45c926535b.png)
2. Lowercase or normalcase will be better
![image](https://user-images.githubusercontent.com/56546832/210239259-70c1ef1c-b266-4947-ba9e-b86a3b3c4cc3.png)
3. I think this "Wyczyść filtry" link should be under "Pokaż meetupy" button and it can be smaller maybe? And trash icon I will put on the left side.
![image](https://user-images.githubusercontent.com/56546832/210239568-f5f0a35b-4224-4526-b046-ea48a55855f7.png) 
 4. If no user is found, button "Pokaż meetupy" shows the amount of all meetups - we want like this?

paulinaszkudlarek commented:
@EwelinaLasowy :)
1, 2,  3, Changed
4. That's right, we do not want it to look like this. I have also changed it, so it shows up only if filters are checked

EwelinaSkrzypacz commented on frontend/components/Meetups/FilterButton.vue:
```diff
@@ -0,0 +1,97 @@
+<template>
+  <div>
+    <Disclosure as="div" class="relative block">
+      <DisclosureButton class="flex w-full cursor-default space-x-3 rounded-md border border-zinc-300 bg-white py-2 px-4 text-center tracking-wider shadow-sm focus:border-zinc-300 focus:outline-none sm:text-base sm:leading-5">
```
```suggestion
      <DisclosureButton class="flex w-full cursor-default space-x-3 rounded-md border border-zinc-300 bg-white py-2 px-4 text-center shadow-sm focus:border-zinc-300 focus:outline-none sm:text-base sm:leading-5">
```

paulinaszkudlarek replied:
@EwelinaLasowy empty state should be working now :)

EwelinaLasowy reviewed (commented):
If there is no meetups to show, we don't have any empty state:
![image](https://user-images.githubusercontent.com/56546832/210331722-64c37b54-8f7d-4646-a6f6-e14cc7d085b3.png)

krzysztofrewak commented on frontend/components/Meetup/useFindNextMeetup.js:
```diff
@@ -7,30 +7,30 @@ onMounted(() => {
     
-    findMeetup()
+    futureMeetups.sort(( a, b ) => {
+      return new Date(a.date).getTime()  -  new Date(b.date).getTime()
```
```suggestion
      return new Date(a.date).getTime() - new Date(b.date).getTime()
```

krzysztofrewak commented on frontend/components/Meetup/useFindNextMeetup.js:
```diff
@@ -7,30 +7,30 @@ onMounted(() => {
     }
     
-    findMeetup()
+    futureMeetups.sort(( a, b ) => {
```
```suggestion
    futureMeetups.sort((a, b) => {
```

krzysztofrewak replied:
Hm, are you running eslint?

paulinaszkudlarek replied:
yes, but I noticed that eslint does not removing all unnecessary whitespaces

krzysztofrewak commented on frontend/components/Meetups/useSortMeetups.js:
```diff
@@ -0,0 +1,55 @@
+export function useSortMeetups (selectedMeetups, sortType) {
```
I think I would not like to have non-components in components directory. Could you move it somewhere else? `Meetup/useFindNextMeetup.js` too.

paulinaszkudlarek replied:
moved :)

krzysztofrewak commented on frontend/components/Meetups/useSortMeetups.js:
```diff
@@ -0,0 +1,55 @@
+  const oldMeetups = []
+  const futureMeetups = []
+  const today = new Date()
+  let sortMeetups = []
```
Maybe `sortedMeetups` would be more verbose? In the name of the function too?

paulinaszkudlarek replied:
you're right, I will change it :)

krzysztofrewak commented on resources/meetups/2023-02-20-laravel-poland-meetup-24-test.json:
```diff
@@ -0,0 +1,26 @@
+{
```
Why `-test` in filename?

krzysztofrewak replied:
Ah, okay. Maybe if it's for testing purposes do not add it to the repository?

paulinaszkudlarek replied:
No problem, I can remove it :)  I wanted you to check how page look when next meetup is set, otherwise everywhere would be empty states

paulinaszkudlarek commented:
@krzysztofrewak @EwelinaLasowy I think a have changed all requested changes, please take a look on that

krzysztofrewak commented on frontend/components/Landing/MeetupIndicator.vue:
```diff
@@ -1,51 +1,39 @@
-    }
-  }
+const computedNextMeetup = computed(() => {
+  if(props.nextMeetup == 0) return {}
```
```suggestion
  if(props.nextMeetup === 0) return {}
```
I would prefer to use strict comparison as often as we can.

krzysztofrewak commented on frontend/components/Meetups/MeetupsList.vue:
```diff
@@ -0,0 +1,134 @@
+})
+
+const findNextMeetup = computed(() => {
+  if (props.allMeetups.length == 0) return {}
```
```suggestion
  if (props.allMeetups.length === 0) return {}
```

krzysztofrewak commented on frontend/components/Meetups/MeetupsList.vue:
```diff
@@ -0,0 +1,134 @@
+
+const meetupSpeakers = ( meetup ) => {
+  let setSpeakers = new Set()
+    meetup.presentations.forEach(({ speakers }) => speakers.forEach(({ name, avatar, image }) => 
+    setSpeakers.add({ name: name, avatar: avatar, image:image }),
+  ))
```
```suggestion
  meetup.presentations.forEach(({ speakers }) => speakers.forEach(({ name, avatar, image }) => 
    setSpeakers.add({ name: name, avatar: avatar, image:image }),
  ))
```

krzysztofrewak commented on frontend/composables/useFindNextMeetup.js:
```diff
@@ -0,0 +1,28 @@
+  const today = new Date()
+  const futureMeetups = []
+
+  const findMeetup = () => {
+    for(let meetup of meetups) {
+      if(new Date(meetup.date) > today) {
+        futureMeetups.push(meetup)
+      }
+    }
+    
+    if(futureMeetups.length > 0) {
+      futureMeetups.sort((a, b) => {
+        return new Date(a.date).getTime() - new Date(b.date).getTime()
+      })
+    }
+    else if(futureMeetups == 0) 
+      return null 
+    
+    nextMeetup.value = futureMeetups[0]
+  }
+  findMeetup()
```
I don't get why this is a separate function that is called one line after its declaration. Do you use this `findMeetup()` anywhere else?

paulinaszkudlarek replied:
it is not used anywhere else, it might be a remnant after some changes. removed :)

krzysztofrewak commented on frontend/composables/useSortedMeetups.js:
```diff
@@ -0,0 +1,55 @@
+
+
+  function sort() {
+    for(const meetup of selectedMeetups) {
+      if(new Date(meetup.date) >= today ) {
+      futureMeetups.push(meetup)
+    }
+    else 
+      oldMeetups.push(meetup)
+    }
+  
+    function compare( a, b ) {
+      if ( a.date < b.date ) {
+        return -1
+      }
+      if ( a.date > b.date ) {
+        return 1
+      }
+      return 0
+    }
```
There are still multiple codestyle issues here like missing curly braces or unneccessary spaces. I will create a issue to take care about it in next PR.

paulinaszkudlarek replied:
I change it now, but eslint does not underline these issues

krzysztofrewak commented on frontend/pages/Landing.vue:
```diff
@@ -5,12 +5,29 @@ import MeetupSpeakers from '@/components/Landing/MeetupSpeakers.vue'
+const meetups = ref([])
+
+const findNextMeetup = computed(() => {
+  if (meetups.value.length == 0) return {}
```
```suggestion
  if (meetups.value.length === 0) return {}
```

krzysztofrewak commented on frontend/pages/Landing.vue:
```diff
@@ -5,12 +5,29 @@ import MeetupSpeakers from '@/components/Landing/MeetupSpeakers.vue'
+})
+
+onMounted(async() => {
+    await fetch('/api/meetups.json').then((response) => response.json()).then((data) => {
+    meetups.value = data
+  })
```
```suggestion
  await fetch('/api/meetups.json').then((response) => response.json()).then((data) => {
    meetups.value = data
  })
```
Also: I suggest to read something like this (https://dmitripavlutin.com/javascript-fetch-async-await/) to understand how await/async and fetch/then works. I would prefer to use one of these approaches.

krzysztofrewak commented on frontend/pages/Landing.vue:
```diff
@@ -5,12 +5,29 @@ import MeetupSpeakers from '@/components/Landing/MeetupSpeakers.vue'
   <hero-section/>
-  <meetup-indicator/>
-  <meetup-speakers/>
+  <meetup-indicator :next-meetup="findNextMeetup"/>
+  <meetup-speakers :next-meetup="findNextMeetup"/>
```
Look, you are calling `findNextMeetup` twice here. Maybe it would be better to call it once in your `script` section and pass result here? Or maybe `MeetupIndicator` and `MeetupSpeakers` could handle different objects after all?

paulinaszkudlarek replied:
Right. I have changed it. I called it now in script section, please take a look if it is optimized

krzysztofrewak reviewed (approved):
:champagne:
